src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -130,6 +249,54 @@ protected LoginInfo initLoginInfo() {
 		return new LoginInfo();
 	}
 
+	/**
+	 * Init instance of QRCCommunicator
+	 *
+	 * @throws Exception if init fail
+	 */
+	public void initQRCCommunicator() throws Exception {
+		qrcCommunicator = new QRCCommunicator();
+		qrcCommunicator.setHost(this.host);
+		qrcCommunicator.setPort(QSYSCoreConstant.QRC_PORT);
+		qrcCommunicator.init();
+	}
+
+	/**
+	 * This method used to update local extended statistics when control a property or in monitoring cycle
+	 *
+	 * @param updateLocalExtStatDto is the dto to update local extended statistic
+	 */
+	private void updateLocalExtStat(UpdateLocalExtStat updateLocalExtStatDto) {
+		if (localExtStat.getStatistics() == null || localExtStat.getControllableProperties() == null) {
+			return;
+		}
+
+		String gainString = updateLocalExtStatDto.getValue();
+		float gainValue = Float.parseFloat(updateLocalExtStatDto.getValue());
+
+		// ex: 9.0 -> 9.00, -2.0 -> -2.00 (Only Gain)
+		if (updateLocalExtStatDto.getControllingMetric() == QSYSCoreControllingMetric.GAIN_VALUE_CONTROL) {
+			if (Math.abs(gainValue / 10) < 1) {
+				gainString = gainString + QSYSCoreConstant.ZERO;
+			}
+			gainString += QSYSCoreConstant.GAIN_UNIT;
+			localExtStat.getStatistics().put(QSYSCoreConstant.GAIN_LABEL + updateLocalExtStatDto.getNamedComponent() + QSYSCoreControllingMetric.CURRENT_GAIN_VALUE.getMetric(), gainString);
+		}
+
+		// Gain or Mute
+		localExtStat.getStatistics().put(updateLocalExtStatDto.getProperty(), "");
+		AdvancedControllableProperty advancedControllableProperty = localExtStat.getControllableProperties().stream()
```

Comment:
would chaining filter -> map work here, instead of extracting new variable explicitly, checking if it exists, setting new value and replacing it?

Reply:
sure, I will do it